src/Microsoft.IdentityModel.Tokens/X509EncryptingCredentials.cs

```diff
@@ -21,7 +23,7 @@ public class X509EncryptingCredentials : EncryptingCredentials
         /// </remarks>
         /// <exception cref="ArgumentNullException">if 'certificate' is null.</exception>
         public X509EncryptingCredentials(X509Certificate2 certificate)
-            : this(certificate, SecurityAlgorithms.DefaultAsymmetricKeyWrapAlgorithm, SecurityAlgorithms.DefaultSymmetricEncryptionAlgorithm)
+            : this(certificate, ShouldUseShortNameForRsaOaepKey() ? SecurityAlgorithms.RsaOAEP : SecurityAlgorithms.DefaultAsymmetricKeyWrapAlgorithm, SecurityAlgorithms.DefaultSymmetricEncryptionAlgorithm)
```

Comment:
You can also extract this `ShouldUseShortNameForRsaOaepKey() ? SecurityAlgorithms.RsaOAEP : SecurityAlgorithms.DefaultAsymmetricKeyWrapAlgorithm,` into a separate method to make this constructor call a bit simpler to read.

```csharp
private static string GetRsaOeapKeyName()
{
return AppContext.TryGetSwitch("Switch.Microsoft.IdentityModel.UseShortNameForRsaOaepKey", out var useKeyWrap) && useKeyWrap ?
SecurityAlgorithms.RsaOAEP :
SecurityAlgorithms.DefaultAsymmetricKeyWrapAlgorithm;
}
```

Reply:
Done. Thanks Peter!